src/codecTools/decodeCodec.ts

```diff
@@ -1,38 +1,46 @@
-// external
-import { fold } from 'fp-ts/lib/Either';
+import { fold, isLeft } from 'fp-ts/lib/Either';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as t from 'io-ts';
 
-// Determine the paths of the codec that are invalid
-const getPaths = <A>(v: t.Validation<A>): string[] =>
-  pipe(
+/**
+ * Determine the codec paths that are invalid.
+ * @param v - Validation errors.
+ * @returns The error strings formatted for human consumption.
+ */
+function getPaths<A>(v: t.Validation<A>): string[] {
+  return pipe(
     v,
     fold(
       (errors) =>
         errors.map((error) => {
-          const lastError = error.context[error.context.length - 1];
-          return `${error.context
-            .map((context) => context.key)
-            // Note: lastError.actual is helpful for debugging
-            .join('.')} expected type '${lastError.type.name}'`;
+          const lastCtx = error.context.at(-1);
+          const fullPath = error.context.map(({ key }) => key).join('.');
+          return `${fullPath} expected type '${lastCtx?.type.name}'`;
```

Comment:
Just checking, it looks like nothing is actually changed here apart from just cleaning it up. But this returns the same thing as before, right? Or am I missing something?